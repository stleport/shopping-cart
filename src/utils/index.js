export const financialFormat = n => `${parseFloat(n).toFixed(2).toString()} â‚¬`;
export const pluralize = n => ((n > 1) ? 's' : '');
