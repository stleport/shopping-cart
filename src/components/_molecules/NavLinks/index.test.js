import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { NavLinks } from '.';
import { pluralize } from '../../../utils';

function NavLinksSetup() {
  const props = {
    cart: {
      items: [{
        productId: 2,
        quantity: 2
      }],
      total: 0
    }
  };

  const NavLinksJSX = (
    <BrowserRouter>
      <NavLinks {...props} />
    </BrowserRouter>
  );

  const NavLinksComponent = mount(NavLinksJSX);
  return {
    NavLinksComponent,
    NavLinksJSX,
    props
  };
}

describe('NavLinks rendering', () => { 
  it('renders correctly NavLinks component', () => {
    const { NavLinksJSX } = NavLinksSetup();
    const NavLinksComponent = renderer.create(NavLinksJSX).toJSON();
    expect(NavLinksComponent).toMatchSnapshot();
  });
});

describe('NavLinks links', () => { 
  const { NavLinksComponent } = NavLinksSetup();

  it('should contain a link to /menu', () => {
    expect(NavLinksComponent.find('a[href="/menu"]').length).toEqual(1);
  });

  it('should contain a link to /cart', () => {
    expect(NavLinksComponent.find('a[href="/cart"]').length).toEqual(1);
  });

  it('should display number of products in cart when the cart is not empty', () => {
    expect(NavLinksComponent.find('a[href="/cart"]').text()).toMatch(/\d+ produits?/);
  });

  it('should pluralize "produit" when more than one product in cart', () => {
    expect(`produit${pluralize(2)}`).toMatch(/^produits$/);
  });
});
