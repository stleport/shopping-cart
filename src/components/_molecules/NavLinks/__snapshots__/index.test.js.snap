// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLinks rendering renders correctly NavLinks component 1`] = `
Array [
  <a
    className="st-Header__item"
    href="/menu"
    onClick={[Function]}
  >
    Menu
  </a>,
  <a
    className="st-Header__item"
    href="/cart"
    onClick={[Function]}
  >
    Panier
     : 1 produit
    
  </a>,
]
`;
